src/integrationTest/java/com/mongodb/kafka/connect/source/MongoSourceTaskIntegrationTest.java

```diff
@@ -579,6 +617,71 @@ void testCopyingExistingWithARestartMidwayThrough() {
     }
   }
 
+  @Test
+  @DisplayName("Copy existing with a restart after finishing")
+  void testCopyingExistingWithARestartAfterFinishing() {
+    try (AutoCloseableSourceTask task = createSourceTask()) {
+
+      MongoCollection<Document> coll = getCollection();
+
+      HashMap<String, String> cfg =
+          new HashMap<String, String>() {
+            {
+              put(MongoSourceConfig.DATABASE_CONFIG, coll.getNamespace().getDatabaseName());
+              put(MongoSourceConfig.COLLECTION_CONFIG, coll.getNamespace().getCollectionName());
+              put(MongoSourceConfig.STARTUP_MODE_CONFIG, StartupMode.COPY_EXISTING.propertyValue());
+              put(MongoSourceConfig.POLL_MAX_BATCH_SIZE_CONFIG, "25");
+              put(MongoSourceConfig.POLL_AWAIT_TIME_MS_CONFIG, "1000");
+            }
+          };
+
+      insertMany(rangeClosed(1, 50), coll);
+      task.start(cfg);
+
+      List<SourceRecord> firstPoll = getNextBatch(task);
+      assertSourceRecordValues(createInserts(1, 25), firstPoll, coll);
+      assertTrue(
+          firstPoll.stream().map(SourceRecord::sourceOffset).allMatch(i -> i.containsKey("copy")));
+
+      List<SourceRecord> secondPoll = getNextBatch(task);
+      assertSourceRecordValues(createInserts(26, 50), secondPoll, coll);
+      // Make sure all elements, except the last one, contains the "copy" key
+      assertTrue(
+          secondPoll.stream()
+              .map(SourceRecord::sourceOffset)
+              .limit(25 - 1) // exclude the last record in the batch
+              .allMatch(i -> i.containsKey("copy")));
+
+      Map<String, ?> lastOffset = secondPoll.get(25 - 1).sourceOffset();
+
+      // Make sure the last copied element does not contain the "copy" key
+      assertFalse(lastOffset.containsKey("copy"));
+
+      // mock the context so that on restart we know where the last task left off
+      when(context.offsetStorageReader()).thenReturn(offsetStorageReader);
+      assertNotNull(lastOffset.get("_id")); // check to make sure the value is an Object
```

Comment:
Same comment as the one above for mocking offset, we could remove this line if we want.